src/Couchbase.Lite.Tests.Shared/ReplicationTest.cs

```diff
@@ -1275,7 +1280,225 @@ public void TestConflictResolverDeletedRemoteWin()
+                return conflict.RemoteDocument;
+            });
+
+            RunReplication(config, 0, 0);
+            
+            wa.WaitForResult(TimeSpan.FromMilliseconds(5000));
+
+            // make sure, first doc starts resolution but finishes last.
+            // in between second doc starts and finishes it.
+            q.ElementAt(0).Should().Be(q.ElementAt(3));
+            q.ElementAt(1).Should().Be(q.ElementAt(2));
+
+            q.Clear();
+        }
+        
+        [Fact]
+        public void TestDoubleConflictResolutionOnSameConflicts()
+        {
+            CreateReplicationConflict("doc1");
+
+            ManualResetEvent manualResetEvent = new ManualResetEvent(false);
+            int resolveCnt = 0;
+
+            var config = CreateConfig(false, true, false);
+            config.ConflictResolver = new TestConflictResolver((conflict) =>
+            {
+                manualResetEvent.Set();
+                Thread.Sleep(500);
+                resolveCnt++;
+                return conflict.LocalDocument;
+            });
+            Replicator replicator = new Replicator(config);
+
+            var config1 = CreateConfig(false, true, false);
+            config1.ConflictResolver = new TestConflictResolver((conflict) =>
+            {
+                resolveCnt++;
+                manualResetEvent.WaitOne();
+                return conflict.RemoteDocument;
+            });
+            Replicator replicator1 = new Replicator(config1);
+
+            _waitAssert = new WaitAssert();
+            var token = replicator.AddChangeListener((sender, args) =>
+            {
+                _waitAssert.RunConditionalAssert(() =>
+                {
+                    VerifyChange(args, 0, 0);
+                    if (config.Continuous && args.Status.Activity == ReplicatorActivityLevel.Idle
+                                          && args.Status.Progress.Completed == args.Status.Progress.Total) {
+                        ((Replicator)sender).Stop();
+                    }
+
+                    return args.Status.Activity == ReplicatorActivityLevel.Stopped;
+                });
+            });
+
+            var _waitAssert1 = new WaitAssert();
+            var token1 = replicator1.AddChangeListener((sender, args) =>
+            {
+                _waitAssert1.RunConditionalAssert(() =>
+                {
+                    VerifyChange(args, 0, 0);
+                    if (config.Continuous && args.Status.Activity == ReplicatorActivityLevel.Idle
+                                          && args.Status.Progress.Completed == args.Status.Progress.Total) {
+                        ((Replicator)sender).Stop();
+                    }
+
+                    return args.Status.Activity == ReplicatorActivityLevel.Stopped;
+                });
+            });
+
+            replicator.Start();
+           
+            Thread.Sleep(50);
+            replicator1.Start();
```

Comment:
Again... hate those Sleeps.  Isn't there a way to do that deterministically?

Reply:
Not this one, it's a straight up hack around the BLIP issue that I mentioned yesterday.